### PR TITLE
test(ali): allow upload from s3 url

### PR DESCRIPTION
**Pull request by MrBatschner**
**How to categorize this PR?**

/kind test
/area os
/os garden-linux

**What this PR does / why we need it**:

Platform tests for Ali-Cloud can now upload the image to test directly from an S3 location to be in line with the other three platforms.
Also adapted the README for the tests so that Cloud Platforms that require a paid subscription are grouped together (and non-paid tests like KVM and CHROOT are grouped together too).

**Release note**:

```feature developer
test(ali): allow upload from s3 url
```

### CODE REVIEW

File: tests/integration/ali.py [hunk 1]
```diff
@@ -177,9 +178,32 @@ def _image_exists(self, image_id):
...
-        
+
+        o = urlparse(image)
+        if o.scheme == "file":
+            logger.info(f"Uploading image {image} - this may take a while...")
+            with open(o.path, "rb") as fileobj:
+                bucket.put_object(image_name_in_bucket, fileobj)
+
+        elif o.scheme == "s3":
+            s3_url = f"https://{o.hostname}.s3.eu-central-1.amazonaws.com/{o.path.lstrip('/')}"
```
Comment (gyptazy) [inline tests/integration/ali.py:189]: Probably we should use `region_id` instead of hardcoded `eu-central-1`

Comment (gyptazy) [reply tests/integration/ali.py:189]: ack, sounds good to me.

**Comment by gyptazy** [comment]
lgtm, just two notes (no stopper for merging):
 * `README.md` - Platforms were in alphabetical order but we may change to an order in priority of platforms
 * Stripping our the hardcoded S3 URI is out of scope of this PR (feel free to adjust this in another one at a later time)

**Review by gyptazy — APPROVED** [review]

**Comment by MrBatschner** [comment]
> lgtm, just two notes (no stopper for merging):
> 
> * `README.md` - Platforms were in alphabetical order but we may change to an order in priority of platforms
> * Stripping out the hardcoded S3 URI is out of scope of this PR (feel free to adjust this in another one at a later time)

Let me address these remarks anyway, so...
/do-not-merge

**Comment by gyptazy** [comment]
> > > lgtm, just two notes (no stopper for merging):
> > > 
> > > * `README.md` - Platforms were in alphabetical order but we may change to an order in priority of platforms
> > > * Stripping out the hardcoded S3 URI is out of scope of this PR (feel free to adjust this in another one at a later time)
> > 
> > 
> > Let me address these remarks anyway, so... /do-not-merge
> 
> Done. /unhold

Thanks for the fast adjustments.

lgtm, feel free to merge.

**Pull request merged**